src/App.ts

```diff
@@ -1540,8 +1547,11 @@ function buildSource<IsEnterpriseInstall extends boolean>(
       // When the app is installed using org-wide deployment, team property will be null
       if (
         typeof bodyAsActionOrOptionsOrViewActionOrShortcut.team !== 'undefined' &&
-        bodyAsActionOrOptionsOrViewActionOrShortcut.team !== null
+        bodyAsActionOrOptionsOrViewActionOrShortcut.team !== null &&
+        'enterprise_id' in bodyAsActionOrOptionsOrViewActionOrShortcut.team
       ) {
+        // TODO: Added null check in if statement, but check
+        // if there's a better way to bypass this Typescript error for enterprise_id
         return bodyAsActionOrOptionsOrViewActionOrShortcut.team.enterprise_id;
```

Comment:
Is the `enterprise_id` a legacy payload thing (ex: from modals, dialogs, etc) that we never deprecated and instead made optional, or is it something where it very much is still being returned in certain legacy payloads? I'm wondering if that's where that logic is from, because I noticed we include that in our test payloads [here](https://github.com/slackapi/bolt-js/blob/main/src/types/options/index.spec.ts#L22-L23). Is there someone that might have context on this? It looks like Kaz was the one that added the unit tests, but this line itself was added years ago by someone who no longer works here. I'm also not able to find any specific documentation that reflects an `enterprise_id` nested under `team`, so I'm not sure if there's just a use case I'm missing where this might be returned, or if it no longer exists anymore as a possibility 🤔

Reply:
Good question, I haven't been around too long myself (just a year and a bit) so perhaps something we bring up in our team meeting to ask @seratch about specifically about this property.

Reply:
@filmaj It seems that this is an incompatibility on the next-gen platform side. Indeed, the new structure is cleaner, though.

Reply:
@seratch do you have any context into whether we should keep this `enterprise_id` check here or update it so that it checks instead for `team.id`, which it seems to always be returning? as of right now we are unable to find a scenario where this `enterprise_id` is being returned under `team`, but want to make sure we're not missing some scenario that would prevent folks from using legacy features!

Also, I'm leaning towards keeping the hybrid structure of the original BlockSuggestions structure + the new next-gen `block_suggestions` structure to maintain legacy compatibility but also open it up to be able to consume the next-gen one, would love to know if you have any thoughts around that especially since it seems that you have experience updating this older BlockSuggestions structure! you can view the updated structure [here](https://github.com/slackapi/bolt-js/pull/1645/files#diff-4d607b6aca75631ef61fa718fc694e67bfae9d4c2d99f69f1124a65876cdad66)!

Reply:
@hello-ashleyintech Legacy apps consistently receives enterprise_id under team object. You can instantly find example payloads in bolt-python tests: https://github.com/slackapi/bolt-python/tree/main/tests/scenario_tests Therefore, this shared logic should not be changed. Regarding the data structure, to avoid future confusions, we may want to have a new separate type for next-gen functions.

Reply:
Also, to make the logic clear, we may want to have a new path to extract remote-function data structure. With this way,  probably we can more easily have two different data types.